File: app/lib/models/item.dart
class Item {
  final int id;
  final String title;
  final String? description;
  // final bool done;
  // final bool public;
  // final ERepeat repeat;
  final DateTime createTime;
  final DateTime modifyTime;

  Item(this.id, this.title, this.description, this.createTime, this.modifyTime);
}

enum ERepeat {
  no,
  everyday,
  weekdays,
  weekends,
  // etc
}

File: web-app/src/models/item.ts
type ItemId = String;
type PlaceId = String;
type OrganizerId = String;
type ParticipantId = String;

enum ERepeat {
  no,
  everyday,
  weekdays,
  weekends,
  // etc
}

export interface IItem {
    id: ItemId,
    title: String,
    description?: String,

    createTime: Date;
    modifyTime: Date;

    from: Date,
    duration: Number,

    done: Boolean,
    public: Boolean,
    repeat: ERepeat,

    place?: PlaceId,
    organizers?: OrganizerId[],
    spots: Number,
    participants: ParticipantId[],
}
